ddtrace/contrib/pytest/plugin.py

```diff
@@ -162,8 +162,9 @@ def pytest_runtest_makereport(item, call):
 
     try:
         result = outcome.get_result()
+        xfail = hasattr(result, "wasxfail") or "xfail" in result.keywords
 
-        if hasattr(result, "wasxfail") or "xfail" in result.keywords:
+        if xfail and not any(x in result.keywords for x in ["skip", "skipif", "skipped"]):
```

Comment:
might be better to extract out `not any(x in result.keywords for x in ["skip", "skipif", "skipped"])` as well, since we use it in multiple places.